### ISSUE TITLE
[Error] sudo asks for root password

### ISSUE THREAD

**Issue by Idesmi** [89luca89/distrobox#1092]
**Describe the bug**
Using `sudo` inside the distrobox containers asks for root password, which is not set and not prompted to be set.

**To Reproduce**
Create rootful container

**Expected behavior**
`sudo` invoked inside the container should ask the user password.

**Logs**
`distrobox-openbuildservice.ini`:
```
[openbuildservice]
additional_packages="quilt spec-cleaner-format_spec_file starship -obs-service-format_spec_file"
image=registry.opensuse.org/opensuse/distrobox-packaging
pull=true
replace=true
root=true
start_now=false
```
`$ distrobox-assemble --verbose create --file ~/distrobox-openbuildservice.ini`
[logs1.txt](https://github.com/89luca89/distrobox/files/13613114/logs1.txt)

I will provide more logs if needed because I am in a hurry.

**Desktop:**
 - podman: 4.8.0
 - distrobox: 1.6.0.1
 - openSUSE MicroOS 20231205

**Comment by pavinjosdev** [comment]
@Idesmi You can set the root password by creating a root shell directly using podman:

```
podman start <name-of-your-container>
podman exec -it <name-of-your-container> bash
# change root password
```

I think this more an image problem and not a distrobox problem as this is not an issue when using toolbx images.

**Comment by mcepl** [comment]
It must be something about SUSE images, because I can reproduce with an image derived from `opensuse/distrobox-packaging:latest`.

**Comment by mcepl** [comment]
@vyskocilm Looking at https://build.opensuse.org/request/show/1130006 makes me wonder something didn’t get broken. Especially the line:

> * Rootful containers now will properly setup user password for sudo

**Comment by vyskocilm** [comment]
@mcepl except the different defaults openSUSE packaging changes nothing. I can reproduce the problem too. I haven't investigated the problem thoroughly. This looks like the change https://github.com/89luca89/distrobox/commit/46b3aa8e651195439666514b450c82dbdc2c4fa2#diff-9020f2495b4d9502f4b28c23de195d9bb5b78f003378f25416bc016cc4996865R1555

And in container logs of `distrobox-enter --root` I have a following

```
+ stat /run/host/etc/shadow
++ stat -c %u /run/host/etc/shadow
+ '[' 0 = 0 ']'
+ '[' '!' -e /run/.nopasswd ']'
+ rootful=1
```

and I am :100: % sure I haven't checked the `--absolutely-disable-root-password-i-am-really-positively-sure` :-) Will try to investigate it more.

**Comment by vyskocilm** [comment]
Never investigate stuff before having the first :coffee: :-) I totally messed up the patch. Reading the description of a change it is clear this is intended _upstream_ change.

https://github.com/89luca89/distrobox/commit/46b3aa8e651195439666514b450c82dbdc2c4fa2

> but is (the passwordless setup) a real concern for rootful containers.
>
> During creation there is a really long and difficult flag that will
> rever to the old behaviour, **this is highly discouraged!**

@89luca8 feel free to close this issue.

**Comment by mcepl** [comment]
> Never investigate stuff before having the first ☕ :-) I totally messed up the patch. Reading the description of a change it is clear this is intended _upstream_ change.

Actually, I am not longing for the previous solution, but for some reason I have my `/etc/sudoers` with this uncommented:

```
Defaults targetpw   # ask for the password of the target user i.e. root
ALL   ALL=(ALL) ALL   # WARNING! Only use this together with 'Defaults targetpw'!
```

Is there some way how to make `sudo` use my users credentials.

**Comment by neurodiverseEsoteric** [comment]
Hey, actually, this has been happening to me on Archlinux, too...

**Comment by deltabluejay** [comment]
Same problem here on OpenSUSE Tumbleweed.

**Comment by fjbaldon** [comment]
I'm not able to run superuser commands on my root distrobox because the password's incorrect. Not sure if they're the same issue though.

**Comment by arvidjaar** [comment]
> Considering
> 
> https://github.com/89luca89/distrobox/blob/3435f4d27070a99668bfa29a3e508db4ecc09009/distrobox-init#L1775
> 
> I thought that `distrobox` should explicitly prevent this from happening?

Tumbleweed sets this default in `/usr/etc/sudoers`, not in `/etc/sudoers`. This file is not supposed to be editable, any overrides should go into `/etc/sudoers.d`.

The same problem would be on any distribution which is using `/etc/sudoers.d` snippets instead of `/etc/sudoers` to set defaults.

**Comment by 89luca89** [comment]
Thanks for the investigation everyone!
I've implemented the workaround in this commit 3b1e312f

**Comment by 1player** [comment]
I am having this problem with archlinux still.

```
toolbox:~ % cat /etc/sudoers.d/sudoers 
Defaults !targetpw
Defaults !fqdn
%wheel ALL=(ALL:ALL) ALL
%sudo ALL=(ALL:ALL) ALL
%root ALL=(ALL:ALL) ALL
```

**Comment by mcepl** [comment]
Yes, this has not been fixed at all. Can still perfectly reproduce with OpenSUSE/Tumbleweed container on MicroOS.

**Comment by sunoc** [comment]
Same issue with a Ubuntu 24.04 on Debian 12.

From inside the container, `/etc/sudoers` is not accessible without root, but `/etc/sudoers.d/sudoers` shows:

```
Defaults !fqdn
"myusername" ALL = (root) NOPASSWD:ALL
```

**Comment by cold-distance** [comment]
Aeon Desktop here. The bug is still there today.

**Comment by archerallstars** [comment]
This issue had already been fixed for openSUSE before Distrobox 1.8.0. After I updated to Distrobox 1.8.0, I can't set the password in newly created root containers again. The return is:

```
⚠️  First time user password setup ⚠️ 
Current password: 
passwd: Authentication token manipulation error
passwd: password unchanged
```

The image is quay.io/fedora/fedora:40

For someone who's facing this issue:

1. `distrobox enter <container-name> -r` to start the container first. Then, disregard the password prompt, and `exit` the container.
2. Enter the container with `sudo podman exec -it <container-name> bash`.
